orderbook/src/maintenance.rs

```diff
@@ -0,0 +1,34 @@
+use crate::database::Database;
+use crate::event_updater::EventUpdater;
+use crate::orderbook::Orderbook;
+use anyhow::Result;
+use futures::try_join;
+use shared::current_block::Maintaining;
+use std::sync::Arc;
+
+pub struct ServiceMaintenance {
```

Comment:
Might be worth adding a comment blurb here (or for the module) explaining that this structure collects all service components that need to run maintenance on every new block.